presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayFindFunction.java

```diff
@@ -0,0 +1,193 @@
+import static com.facebook.presto.type.TypeUtils.parameterizedTypeName;
+import static com.facebook.presto.type.TypeUtils.readStructuralBlock;
+import static com.facebook.presto.util.Reflection.methodHandle;
+
+public final class ArrayFindFunction
+        extends ParametricScalar
+{
+    public static final ArrayFindFunction ARRAY_FIND = new ArrayFindFunction();
+    private static final Signature SIGNATURE = new Signature("array_find", ImmutableList.of(typeParameter("E")), "bigint", ImmutableList.of("E", "array<E>"), false, false);
+    private static final MethodHandle METHOD_HANDLE_BOOLEAN = methodHandle(ArrayFindFunction.class, "arrayFind", Type.class, MethodHandle.class, boolean.class, Slice.class);
+    private static final MethodHandle METHOD_HANDLE_LONG = methodHandle(ArrayFindFunction.class, "arrayFind", Type.class, MethodHandle.class, long.class, Slice.class);
+    private static final MethodHandle METHOD_HANDLE_DOUBLE = methodHandle(ArrayFindFunction.class, "arrayFind", Type.class, MethodHandle.class, double.class, Slice.class);
+    private static final MethodHandle METHOD_HANDLE_SLICE = methodHandle(ArrayFindFunction.class, "arrayFind", Type.class, MethodHandle.class, Slice.class, Slice.class);
+
+    @Override
+    public Signature getSignature()
+    {
+        return SIGNATURE;
+    }
+
+    @Override
+    public boolean isHidden()
+    {
+        return false;
+    }
+
+    @Override
+    public boolean isDeterministic()
+    {
+        return false;
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return null;
+    }
+
+    @Override
+    public FunctionInfo specialize(Map<String, Type> types, int arity, TypeManager typeManager, FunctionRegistry functionRegistry)
+    {
+        Type type = types.get("E");
+        MethodHandle equalMethodHandle = functionRegistry.resolveOperator(OperatorType.EQUAL, ImmutableList.of(type, type)).getMethodHandle();
+        MethodHandle arrayFindMethodHandle;
+        if (type.getJavaType() == boolean.class) {
+            arrayFindMethodHandle = METHOD_HANDLE_BOOLEAN;
+        }
+        else if (type.getJavaType() == long.class) {
+            arrayFindMethodHandle = METHOD_HANDLE_LONG;
+        }
+        else if (type.getJavaType() == double.class) {
+            arrayFindMethodHandle = METHOD_HANDLE_DOUBLE;
+        }
+        else if (type.getJavaType() == Slice.class) {
+            arrayFindMethodHandle = METHOD_HANDLE_SLICE;
+        }
+        else {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Argument type to array_find unsupported");
+        }
+        return new FunctionInfo(
+                new Signature("array_find", parseTypeSignature(StandardTypes.BIGINT), type.getTypeSignature(), parameterizedTypeName("array", type.getTypeSignature())),
+                "Returns the cardinality (length) of the array",
+                false,
+                arrayFindMethodHandle.bindTo(type).bindTo(equalMethodHandle),
+                true,
```

Comment:
use `isHidden()` and `isDeterministic()` to make sure these stay in sync and to make it easier to read